### PR TITLE
feat: allow spend L1 consensus trigger

### PR DESCRIPTION
**Pull request by marcinwadon**

### CODE REVIEW

File: modules/currency-l1/src/main/scala/io/constellationnetwork/currency/l1/http/SwapRoutes.scala [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    // TODO: @mwadon - validation
+    onValid
+
+  protected val public: HttpRoutes[F] = HttpRoutes.of[F] {
+    case GET -> Root / "swap" =>
+      swapTransactionsQueue.view.flatMap(Ok(_))
+
+    case req @ POST -> Root / "swap" =>
+      req
+        .asR[Signed[SwapTransaction]] {
```
Comment (scasplte2) [inline modules/currency-l1/src/main/scala/io/constellationnetwork/currency/l1/http/SwapRoutes.scala:55]: Perhaps I am still misunderstanding but I thought the SwapTransaction was to be treated like Rewards and not generated by the end-user? Doesn't this structure require the user to sign and generate the swap transaction?

Comment (scasplte2) [reply modules/currency-l1/src/main/scala/io/constellationnetwork/currency/l1/http/SwapRoutes.scala:55]: Perhaps the `SwapTransaction` is what was previously called AllowSpend? In that case, I understand. I was thinking this is the balance updating transaction and not the "user authorization" portion of the process.

Comment (marcinwadon) [reply modules/currency-l1/src/main/scala/io/constellationnetwork/currency/l1/http/SwapRoutes.scala:55]: Yes, it's a trait above the AllowSpend. @AlexBrandes wanted to make it possible to put new types of transactions in the future. Maybe SwapTransaction is a bit unfortunate name, we can think of something better.

Comment (marcinwadon) [reply modules/currency-l1/src/main/scala/io/constellationnetwork/currency/l1/http/SwapRoutes.scala:55]: I think we should definitely refine and write down naming conventions for AMM 🤔

Comment (scasplte2) [reply modules/currency-l1/src/main/scala/io/constellationnetwork/currency/l1/http/SwapRoutes.scala:55]: yes, I was able to find it [here](https://github.com/Constellation-Labs/tessellation/blob/8dde3dd93cae65bb17d77fd7ca47e797962c8cac/modules/shared/src/main/scala/io/constellationnetwork/schema/swap.scala#L88). Thanks for clarifying as well.

**Review by scasplte2 — APPROVED** [review]

**Review by IPadawans — APPROVED** [review]

**Review by scasplte2 — APPROVED** [review]

**Pull request merged**